gcovr/tests/add_coverages/reference/coverage.json

```diff
@@ -0,0 +1,167 @@
+{
+    "gcovr/format_version": 0.1, 
```

Comment:
This file contains a space after the comma, which causes all tests to fail except those under 2.7. There are two alternatives how this can be resolved:

* in the test runner, a JSON scrubber is added that trims all lines to remove the trailing space if any is present
* in the test runner, JSON files aren't compared textually but the JSON is loaded into a data structure that is then compared, similar to the current XML tests. This could be done in a scrubber or in a custom ASSERT_EQUALS.

Reply:
I can explicit set separators in json.dump, like here:

```diff
--- a/gcovr/json_generator.py
+++ b/gcovr/json_generator.py
@@ -37,7 +37,7 @@ def print_json_report(covdata, output_file, options):
 
     write_json = json.dump
     if options.prettyjson:
-        write_json = functools.partial(write_json, indent=PRETTY_JSON_INDENT)
+        write_json = functools.partial(write_json, indent=PRETTY_JSON_INDENT, separators=(',', ': '))
 
     if options.output:
         with open(options.output, 'w') as output:
```

JSON output in pretty-print mode will be without space at the end of line with comma, what can be useful if somebody wants to parse JSON file with external custom tool. Space at the end of line with comma can be annoying.

> 
- in the test runner, JSON files aren't compared textually but the JSON is loaded into a data structure that is then compared, similar to the current XML tests. This could be done in a scrubber or in a custom ASSERT_EQUALS.

It can help to pass test when one the file is in pretty-print mode and second without it, so I have to implement this comparison.

Reply:
Oh, I didn't know you could change the separators! If setting that option fixes the problem, that seems like the easiest solution. Do whatever *you* think is best, as long as the tests pass on Python 2.7 and 3.6.

Reply:
In my last PR I have added `separators=(',', ': ')` and `sort_keys=True` to the `json.dump()`.
Dictionaries in python are not ordered, therefore `coverage data` can be generated in different ways in JSON output files. `sort_keys=True` preserves alphanumeric ordering for dictionaries used in `json.dump()`. Content of output files will be generated always in the same order what is useful to compare JSON output file like a plain text files during tests.

I think thanks to that tests for Python 2.7 and 3.6 were passed.

Actually I though about comparing JSON structures during tests after gathering information from `json.load()`, convert this to the `coverage data` and next add `compare_coverages()` function to utils.py and use it, but maybe is better to differentiate between pretty-print and non pretty-print mode and compare raw text files. After conversion from JSON file to `coverage` we will lose information about pretty printing. What if in the future something goes wrong with pretty-print during adding some new features. `coverage data` will be ok, but pretty-print can be damaged without information about that during test.
It is always decision to make, usage of `separators=(',', ': ')` and `sort_keys=True` looks like to be the easiest one.